ui/src/components/LoginForm/utils.js

```diff
@@ -0,0 +1,26 @@
+function parseDate(yyyymmdd) {
+  const year = yyyymmdd.slice(0, 4);
+  const month = yyyymmdd.slice(4, 6);
+  const day = yyyymmdd.slice(6, 8);
+  return new Date(`${year}-${month}-${day}`);
+}
+
+export function isDateInRange(startDateStr, endDateStr, rangeInDays) {
+  // Helper function to parse YYYYMMDD format into Date object
```

Comment:
Should be moved or removed (I think removed is fine, as the function `parseDate` is clear enough IMO).